erc20-bridge-token/contracts/BridgeTokenFactory.sol

```diff
@@ -0,0 +1,152 @@
+pragma solidity ^0.8;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/math/SafeMathUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+
+import "rainbow-bridge-sol/nearprover/contracts/INearProver.sol";
+import "rainbow-bridge-sol/nearprover/contracts/ProofDecoder.sol";
+import "rainbow-bridge-sol/nearbridge/contracts/NearDecoder.sol";
+import "rainbow-bridge-sol/nearbridge/contracts/Borsh.sol";
+
+import "./Locker.sol";
+import "./BridgeToken.sol";
+import "./BridgeTokenProxy.sol";
+
+contract BridgeTokenFactory is Locker, AccessControlUpgradeable, PausableUpgradeable{
```

Comment:
Please add scripts for deploying and upgrading the factory using the proxy.

Reply:
Last version of scripts: bca251aebaa46363772452f237cad88e58bfc364